### PR TITLE
Reset password logic

### PR DESCRIPTION
**Pull request by banguyenvareal**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  def get_user
+    @user = User.find_by(password_reset_token: params[:id])
+    set_flash_and_redirect(:danger, t('users.not_exist'), root_path) unless
+      @user
+  end
+
+  def password_match?
+    password_confirmation = params[:user][:password_confirmation]
+    password = params[:user][:password]
+    !password_confirmation.eql?(password)
```
Comment (dinhvareal) [inline app/controllers/password_resets_controller.rb]: `password_confirmation.eql?(password)`

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  before_action :check_user_exist, only: %i[create]
+
+  def create
+    @user.send_password_reset(@user.phone, params[:password_reset][:email])
+    set_flash_and_redirect(:info, t('password_reset.notify'), root_path)
+  end
+
+  def update
+    if password_match?
+      flash[:danger] = t('password_reset.password_not_match')
```
Comment (dinhvareal) [inline app/controllers/password_resets_controller.rb]: let it on `password_match` and rename this method

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      render 'edit'
+    elsif @user.update_attributes(user_params)
+      @user.update_attribute(:password_reset_token, nil)
+      set_flash_and_redirect(:success, t('password_reset.sucess'), root_path)
+    else
+      render 'edit'
```
Comment (dinhvareal) [inline app/controllers/password_resets_controller.rb]: add flash

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
```
Comment (dinhvareal) [inline config/routes.rb]: remove

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  def get_user
+    @user = User.find_by(password_reset_token: params[:id])
+    set_flash_and_redirect(:danger, t('users.not_exist'), root_path) unless
+      @user
+  end
+
+  def check_password_match
+    password_confirmation = params[:user][:password_confirmation]
```
Comment (dinhvareal) [inline app/controllers/password_resets_controller.rb]: `user_params[:password_confirmation]`

File: app/controllers/password_resets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    password = params[:user][:password]
```
Comment (dinhvareal) [inline app/controllers/password_resets_controller.rb]: `user_params[:password]`

**Review by hieunguyentrung — APPROVED** [review]

**Pull request merged**